yarn-project/simulator/src/avm/journal/journal.ts

```diff
@@ -352,33 +354,41 @@ export class AvmPersistableStateManager {
     const siloedNullifier = siloNullifier(contractAddress, nullifier);
 
     if (this.doMerkleOperations) {
+      // Maybe overkill, but we should check if the nullifier is already present in the tree before attempting to insert
+      // It might be better to catch the error from the insert operation
       // Trace all nullifier creations, even duplicate insertions that fail
-      const alreadyPresent = await this.merkleTrees.getPreviousValueIndex(
+      const { preimage, index, update } = await this.merkleTrees.getLeafOrLowLeafInfo(
         MerkleTreeId.NULLIFIER_TREE,
-        siloedNullifier.toBigInt(),
+        siloedNullifier,
       );
-      if (alreadyPresent) {
-        this.log.verbose(`Nullifier already present in tree: ${nullifier} at index ${alreadyPresent.index}.`);
+      if (update) {
+        this.log.verbose(`Nullifier already present in tree: ${nullifier} at index ${index}.`);
+        // If the nullifier is already present, we should not insert it again
```

Comment:
Hmm I don't think this function is working properly right now. On collision, `this.nullifiers.append()` above will throw an error and user code will revert. I think we want to trace the membership check of the already-present nullifier and then rethrow?